### ISSUE TITLE
"Cannot prove `root` is not nil"

### ISSUE THREAD

**Issue by dawkot** [pragmagic/karax#138]
## Code
```nim
include karax/prelude
```
## Output
```
/home/dawkot/.nimble/pkgs/karax-#head/karax/karax.nim(679, 34) Error: cannot prove 'root' is not nil
```
## Other
* Tested with Nim 1.2.0 and some previous devels
* Compiles with 1.0.6

**Comment by Araq** [comment]
Works for me, update your karax version?

**Comment by dawkot** [comment]
Ok, it works with #head. I thought I tried that.

**Comment by FedericoCeratto** [comment]
Same for me with karax@1.1.0 and Nim 1.2.0

**Comment by Araq** [comment]
@FedericoCeratto But I released Karax 1.1.1.
